src/engine_registry.cpp

```diff
@@ -45,7 +47,28 @@ namespace wr
 		{
 			{ TextureFilter::FILTER_POINT, TextureAddressMode::TAM_BORDER }
 		}
-	});
+		});
+
+	//MipMapping Root Signature
+	std::array< CD3DX12_DESCRIPTOR_RANGE, 1> mip_in_srv_ranges
+	{
+		[] { CD3DX12_DESCRIPTOR_RANGE r; r.Init(D3D12_DESCRIPTOR_RANGE_TYPE_SRV, 1, 0, 0); return r; } ()
+	};
+	std::array< CD3DX12_DESCRIPTOR_RANGE, 1> mip_out_uav_ranges
+	{
+		[] { CD3DX12_DESCRIPTOR_RANGE r; r.Init(D3D12_DESCRIPTOR_RANGE_TYPE_UAV, 4, 0, 0); return r; } ()
+	};
+	REGISTER(root_signatures::mip_mapping) = RootSignatureRegistry::Get().Register({
+		{
+			[] { CD3DX12_ROOT_PARAMETER d; d.InitAsConstants(2, 0); return d; }(),
+			[] { CD3DX12_ROOT_PARAMETER d; d.InitAsDescriptorTable(mip_in_srv_ranges.size(), mip_in_srv_ranges.data()); return d; }(),
```

Comment:
You can replace this with a single descriptor table.

Reply:
Done that, thank you 👍